intranet/static/js/files.js

```diff
@@ -1,91 +1,96 @@
+                    if (dt.files.length) {
+                        var count = dt.files.length;
+                        $.each(dt.files, function(index, file) {
+                            var safeName = $("<div>").text(file.name).html();
+                            var message = Messenger().post({
+                                "message": "Uploading <b>" + safeName + "</b>",
+                                "type": "info"
+                            });
+                            var fd = new FormData();
+                            fd.append("file", file);
+                            fd.append("csrfmiddlewaretoken", $.cookie("csrftoken"));
+                            $.ajax({
+                                url: endpoint,
+                                type: "POST",
+                                contentType: false,
+                                processData: false,
+                                cache: false,
+                                data: fd,
+                                success: function(data) {
+                                    message.update({
+                                        "message": "<b>" + safeName + "</b> uploaded",
+                                        "type": "success"
+                                    });
+
+                                    count--;
+
+                                    if (count <= 0) {
+                                        if ($("#directory-list").length) {
+                                            var dirList = $("#directory-list", $(data)).html();
+                                            $("#directory-list").html(dirList);
+                                        }
                                     }
+                                },
+                                error: function(xhr, stat, err) {
+                                    message.update({
+                                        "message": "Failed to upload <b>" + safeName + "</b>",
+                                        "type": "error"
+                                    });
                                 }
-                            },
-                            error: function(xhr, stat, err) {
-                                message.update({
-                                    "message": "Failed to upload <b>" + safeName + "</b>",
-                                    "type": "error"
-                                });
-                            }
+                            });
                         });
-                    });
-                    e.preventDefault();
+                        e.preventDefault();
+                    }
                 }
             }
         });
-    }
 
+        $("#upload-overlay").on("dragleave", function(e) {
+            $("#upload-overlay").stop().fadeOut(150);
+        });
+    }
 
-    zip_folder = function(ths) {
-        var name = $(this).attr("data-name");
+    zip_folder = function(item) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["zip_folder" is not defined.](https://www.codacy.com/app/tjhsst-ion/ion/file/2769576146/issues/source?bid=3212528&fileBranchId=3355681#l83)